File: src/Backend/Inventory.Data/Models/Product/ProductVariantModel.cs
namespace Inventory.Data.Models.Product
{
    public class ProductVariantModel
    {
        public int VariantId { get; set; }

        public string VariantName { get; set; }

        public int ProductId { get; set; }

        public string ProductName { get; set; }

        public string SKU { get; set; }

        public string BarCode { get; set; }

        public string Attributes { get; set; }

        public decimal StockQuantity { get; set; }

        public float RetailPrice { get; set; }

        public float MaxDiscount { get; set; }
    }
}

File: src/Frontend/src/app/dashboard/Models/product-variant.model.ts
export class ProductVariantModel {
    /**
     * Variant Id
     * @type {number}
     */
    variantId: number;

    /**
     * Variant Name
     * @type {string}
     */
    variantName: string;

    /**
     * Quantity of the variant currently available in stock
     * @type {number}
     */
    stockQuantity: number;

    /**
     * Stock Keeping Unit
     * @type {string}
     */
    sku: string = '';

    /**
     * Attributes of the variant
     * @type {string}
     */
    attributes: string = '';

    /**
     * Product Id of the product this variant belongs to
     * @type {number}
     */
    productId: number;
    /**
     * Product purchase price
     * @type {number}
     */
    purchasePrice: number = 0;
    /**
     * Product retail price
     * @type {number}
     */
    retailPrice: number = 0;
    /**
     * Purchase/Sales/Inventory Quantity
     * @type {number}
     */
    quantity: number = 0;
}